Source/ASCollectionView.mm

```diff
@@ -2311,7 +2311,11 @@ - (void)rangeController:(ASRangeController *)rangeController updateWithChangeSet
       
       // Flush any range changes that happened as part of submitting the update.
       as_activity_scope(changeSet.rootActivity);
-      [self->_rangeController updateIfNeeded];
+      if (numberOfUpdates > 0 && ASActivateExperimentalFeature(ASExperimentalRangeUpdateOnChangesetUpdate)) {
+        [self->_rangeController setNeedsUpdate];
+      } else {
+        [self->_rangeController updateIfNeeded];
```

Comment:
Do we also want to run `updateIfNeeded` right after `setNeedsUpdate` in the experiment? I don't think it's a big deal either ways because `updateIfNeeded` will be called shortly by other call sites (like the one in the `completion` block [here](https://github.com/TextureGroup/Texture/pull/1976/files#diff-a763091124ee49f35c39703ce93f1d38fb204a058ff6483896416ffa678aa155R2258)), but wrapping it in an `else` statement seems like an unnecessary change for this experiment?

```
      if (numberOfUpdates > 0 && ASActivateExperimentalFeature(ASExperimentalRangeUpdateOnChangesetUpdate)) {
        [self->_rangeController setNeedsUpdate];
      }
      
      [self->_rangeController updateIfNeeded];
```

Reply:
Originally I put it in the `else` because it would get dispatched in a call within `setNeedsUpdate` [here](https://github.com/TextureGroup/Texture/blob/5a205d84c842c8709819898bd6ee4a74c1f91a3d/Source/Details/ASRangeController.mm#L139), but now I actually think it's better to leave the `updateIfNeeded` call just in case the range controller's `_needsRangeUpdate` is already set to `YES` and the call to `updateIfNeeded` is not dispatched. I'll update this